kolibri/utils/tests/test_handler.py

```diff
@@ -28,3 +31,23 @@ def test_do_rollover(self):
         settings.LOGGING["handlers"]["file"]["when"] = orig_value
 
         self.assertNotEqual(os.listdir(archive_dir), [])
+
+    def test_getFilesToDelete(self):
+        temp_dir = tempfile.mkdtemp()
+        file_handle, log_file = tempfile.mkstemp(suffix=".txt", dir=temp_dir)
```

Comment:
It's a pitty we don't have TemporaryDirectory in Python 2.7, but, for the purpose of this code, could it be done using` with tempfile.TemporaryFile() as log_file` , so you don't need to import and use shutil?
i.e. 
```diff
@@ -1,5 +1,4 @@
 import os
-import shutil
 import tempfile
 from time import sleep
 
@@ -33,21 +32,14 @@ class KolibriTimedRotatingFileHandlerTestCase(TestCase):
         self.assertNotEqual(os.listdir(archive_dir), [])
 
     def test_getFilesToDelete(self):
-        temp_dir = tempfile.mkdtemp()
-        file_handle, log_file = tempfile.mkstemp(suffix=".txt", dir=temp_dir)
-        os.close(file_handle)
-        handler = KolibriTimedRotatingFileHandler(log_file, backupCount=3, when="s")
-        sleep(1)
-        handler.doRollover()
-        sleep(1)
-        handler.doRollover()
-        sleep(1)
-        handler.doRollover()
-        sleep(1)
-        handler.doRollover()
-        self.assertEqual(len(handler.getFilesToDelete()), 1)
-        try:
-            os.remove(log_file)
-        except OSError:
-            pass
-        shutil.rmtree(temp_dir, ignore_errors=True)
+        with tempfile.NamedTemporaryFile(suffix=".txt") as log_file:
+            handler = KolibriTimedRotatingFileHandler(log_file.name, backupCount=3, when="s")
+            sleep(1)
+            handler.doRollover()
+            sleep(1)
+            handler.doRollover()
+            sleep(1)
+            handler.doRollover()
+            sleep(1)
+            handler.doRollover()
+            self.assertEqual(len(handler.getFilesToDelete()), 1)

```

Reply:
No, I've had to remove our use of `NamedTemporaryFile` in general, because it breaks on Windows due to permissions errors.